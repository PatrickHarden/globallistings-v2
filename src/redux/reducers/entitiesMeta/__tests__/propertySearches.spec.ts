import payload from '../../../../tests/fixtures/residential-response-normalised.json';

import reducer from '../propertySearches';

const MOCK_HASH = 'MOCK_HASH';
const MOCK_HASH_2 = 'MOCK_HASH_2';
const CREATED_AT_MOCK = new Date('2018-12-10T11:11:13Z');

describe('handle SUCCESS action', () => {
  const p1 = {
    entities: {
      properties: {
        P1: {
          title: 'title 1',
        },
      },
    },
    result: ['P1'],
  };

  const p2 = {
    entities: {
      properties: {
        P2: {
          title: 'title 2',
        },
      },
    },
    result: ['P2'],
  };

  const p1Modified = {
    entities: {
      properties: {
        P1: {
          title: 'title 1 modified',
        },
      },
    },
    result: ['P1'],
  };

  it('should correctly handle single SUCCESS', () => {
    const successAction = {
      payload,
      type: 'SEARCH_PROPERTIES_SUCCESS',
      meta: { hash: MOCK_HASH, receivedAt: CREATED_AT_MOCK },
    };
    const newState = reducer({}, successAction as any);

    expect(newState).toMatchSnapshot();
  });

  it('should amend old property searches with new', () => {
    const successAction1 = {
      payload: p1,
      type: 'SEARCH_PROPERTIES_SUCCESS',
      meta: { hash: MOCK_HASH, receivedAt: CREATED_AT_MOCK },
    };
    const successAction2 = {
      payload: p2,
      type: 'SEARCH_PROPERTIES_SUCCESS',
      meta: { hash: MOCK_HASH_2, receivedAt: CREATED_AT_MOCK },
    };
    const newState1 = reducer({}, successAction1 as any);
    const newState2 = reducer(newState1, successAction2 as any);

    expect(newState2).toMatchSnapshot();
  });

  it('should overwrite old properties with same id', () => {
    const successAction1 = {
      payload: p1,
      type: 'SEARCH_PROPERTIES_SUCCESS',
      meta: { hash: MOCK_HASH, receivedAt: CREATED_AT_MOCK },
    };
    const successAction2 = {
      payload: p1Modified,
      type: 'SEARCH_PROPERTIES_SUCCESS',
      meta: { hash: MOCK_HASH, receivedAt: CREATED_AT_MOCK },
    };
    const newState1 = reducer({}, successAction1 as any);
    const newState2 = reducer(newState1, successAction2 as any);

    expect(newState2).toMatchSnapshot();
  });
});

it('should correctly handle request', () => {
  const successAction = {
    type: 'SEARCH_PROPERTIES_REQUEST',
    meta: { hash: MOCK_HASH, receivedAt: CREATED_AT_MOCK },
  };
  const newState = reducer({}, successAction as any);

  expect(newState).toMatchSnapshot();
});

it('should correctly handle failure', () => {
  const successAction = {
    type: 'SEARCH_PROPERTIES_FAILURE',
    payload: new Error('boom'),
    meta: {
      hash: MOCK_HASH,
      receivedAt: CREATED_AT_MOCK,
    },
  };
  const newState = reducer({}, successAction as any);

  expect(newState).toMatchSnapshot();
});
