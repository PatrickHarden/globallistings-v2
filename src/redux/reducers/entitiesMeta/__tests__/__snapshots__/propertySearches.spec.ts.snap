// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle SUCCESS action should amend old property searches with new 1`] = `
Object {
  "MOCK_HASH": Object {
    "createdAt": 2018-12-10T11:11:13.000Z,
    "status": "SUCCESS",
  },
  "MOCK_HASH_2": Object {
    "createdAt": 2018-12-10T11:11:13.000Z,
    "status": "SUCCESS",
  },
}
`;

exports[`handle SUCCESS action should correctly handle single SUCCESS 1`] = `
Object {
  "MOCK_HASH": Object {
    "createdAt": 2018-12-10T11:11:13.000Z,
    "status": "SUCCESS",
  },
}
`;

exports[`handle SUCCESS action should overwrite old properties with same id 1`] = `
Object {
  "MOCK_HASH": Object {
    "createdAt": 2018-12-10T11:11:13.000Z,
    "status": "SUCCESS",
  },
}
`;

exports[`should correctly handle failure 1`] = `
Object {
  "MOCK_HASH": Object {
    "error": [Error: boom],
    "status": "ERROR",
  },
}
`;

exports[`should correctly handle request 1`] = `
Object {
  "MOCK_HASH": Object {
    "status": "PENDING",
  },
}
`;
